### PR TITLE
Added siddhi apps for throttledout,fault statistics and other types o…

### PR DESCRIPTION
**Pull request by shilmyhasan**
added Siddhi apps for ThrottledOut and Fault streams and also added siddhi apps for alerts HealthAvailability,FrequentTierLimitHitting and Unusual IP access.
note ; Made a change to ApiRequestProcessor app to make a table join as left outer join.

### CODE REVIEW

File: modules/distribution/carbon-home/deployment/siddhi-files/APIM_FREQUENT_TIER_LIMIT_HITTING_ALERT.siddhi [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+@info(name='Email generation query') 
+from TierLimitHittingAlertStream#window.length(1) as TL join ApimAlertStakeholderInfo as ASI
+on ((TL.subscriber == ASI.userId and true == ASI.isSubscriber ) or true == ASI.isAdmin )
+select 'FrequentTierHittingAlert' as type ,
+ifThenElse(str:contains(TL.message, 'Application frequently goes beyond the allocated quota'),str:concat("The application ", TL.applicationName, " owned by ", TL.subscriber, " frequently goes beyond the allocated quota when accessing the ", TL.apiName," API."),
+str:concat(TL.message , " Using the ", TL.applicationName, " application owned by ",TL.subscriber, ".")) as message, time:dateFormat(TL.alertTimestamp,'yyyy-MM-dd HH:mm:ss') as alertTimestamp,ASI.emails
+having  str:contains(ASI.alertTypes, 'FrequentTierHittingAlert')
+insert into EmailAlertStream;
+
+
```
Comment (chamilaadhi) [inline modules/distribution/carbon-home/deployment/siddhi-files/APIM_FREQUENT_TIER_LIMIT_HITTING_ALERT.siddhi]: shall we remove this whitespace

Comment (shilmyhasan) [reply modules/distribution/carbon-home/deployment/siddhi-files/APIM_FREQUENT_TIER_LIMIT_HITTING_ALERT.siddhi]: done

File: modules/distribution/carbon-home/deployment/siddhi-files/APIM_FAULT_SUMMARY.siddhi [hunk 2]
```diff
@@ -0,0 +1,32 @@
+@App:name("APIM_FAULT_SUMMARY")
```
Comment (ruks) [inline modules/distribution/carbon-home/deployment/siddhi-files/APIM_FAULT_SUMMARY.siddhi]: can we add a license header if possible? also check the other files as well.

Comment (dinusha92) [reply modules/distribution/carbon-home/deployment/siddhi-files/APIM_FAULT_SUMMARY.siddhi]: @shilmyhasan can you verify if the license headers have been added as requested?. (PR or commit)

Comment (shilmyhasan) [reply modules/distribution/carbon-home/deployment/siddhi-files/APIM_FAULT_SUMMARY.siddhi]: yes, Its already added in 2bc9dabb511a5de2819c5ddf1af81cbe6c3e46cb

**Pull request merged**